src/ObjectList.php

```diff
@@ -195,6 +196,19 @@ public function exclude(ObjectList $otherList)
     }
 
     /**
+     * Return a new list containing only those elements present in {$otherList}.
+     *
+     * @param ObjectList $otherList
+     * @return static
+     */
+    public function retain(ObjectList $otherList)
```

Comment:
`intersect` or `intersection` is the name of this method I found in collection libs. https://en.wikipedia.org/wiki/Intersection_(set_theory)